services/blockchain-indexer/shared/dataService/business/blocks.js

```diff
@@ -53,31 +55,46 @@ const normalizeBlock = async (originalblock) => {
 
 	if (block.generatorAddress) {
 		block.generatorAddress = await getBase32AddressFromHex(block.generatorAddress);
+
+		const generatorInfo = await getIndexedAccountInfo(
+			{ address: block.generatorAddress, limit: 1 },
+			['address', 'publicKey', 'name'],
+		);
+
+		block.generator = {
+			address: block.generatorAddress,
+			publicKey: generatorInfo ? generatorInfo.publicKey : null,
+			name: generatorInfo ? generatorInfo.name : null,
+		};
 	}
 
+	// TODO: Enable when events are available
+	// block.events = await getEventsByHeight(block.height);
+
 	block.isFinal = block.height <= (await getFinalizedHeight());
 	block.numberOfTransactions = block.transactions.length;
+	block.numberOfAssets = block.assets.length;
+	block.numberOfEvents = 1;
+	// block.numberOfEvents = block.events.length;
```

Comment:
Please avoid fetching the events here. Instead, please index the number of events per block and return the count.